### PR TITLE
Add User-Agent and escape "\n" when serializing JSON strings to NQuads

### PR DESCRIPTION
**Pull request by frmichel**
Hi Markus,

This is a 2nd pull request which includes #80 and adds 2 more changes:

- add a User-Agent HTTP header when loading document by URL in FileGetContentsLoader. I noticed a Web API that fails if there is none. I set "User-Agent: JsonLD.php"
- additional pb with when serializing JSON-LD to NQuands: if a JSON string contains "\n" then the triples generated are badly interpreted. I simply removed the "\n# in that case.

Cheers,
   Franck.

### CODE REVIEW

File: FileGetContentsLoader.php [hunk 1]
```diff
@@ -31,8 +31,12 @@ public function loadDocument($url)
 
             $streamContextOptions = array(
               'method'  => 'GET',
-              'header'  => "Accept: application/ld+json, application/json; q=0.9, */*; q=0.1\r\n",
-              'timeout' => Processor::REMOTE_TIMEOUT
+              'header'  => "Accept: application/ld+json, application/json; q=0.9, */*; q=0.1\r\n".
+                           // Fix F. Michel 2017-12-06 - Some Web API require a User-Agent.
+                           // E.g. MusicBrainz returns error 403 if there is none.
+                           "User-Agent: JsonLD.php\r\n",
```
Comment (lanthaler) [inline FileGetContentsLoader.php]: Please change the user agent name, remove the comments and move the dot on to the next line

```
              'header'  => "Accept: application/ld+json, application/json; q=0.9, */*; q=0.1\r\n"
                             . "User-Agent: JsonLD\r\n",
```

File: FileGetContentsLoader.php [hunk 2]
```diff
@@ -31,8 +31,12 @@ public function loadDocument($url)
...
+              'timeout' => Processor::REMOTE_TIMEOUT,
+              'ssl' => [ 'verify_peer' => true, 'verify_peer_name' => true, 'allow_self_signed'=> false ]
```
Comment (lanthaler) [inline FileGetContentsLoader.php]: If you want the SSL change to be included, please file a separate PR. I think it needs more discussion

File: JsonLD.php [hunk 3]
```diff
@@ -171,6 +172,80 @@ public static function expand($input, $options = null)
         return $input;
     }
 
+
```
Comment (lanthaler) [inline JsonLD.php]: All the changes in this file are completely unrelated to the PR. Please revert them.

File: NQuads.php [hunk 4]
```diff
@@ -41,7 +41,11 @@ public function serialize(array $quads)
                     ? $quad->getObject()
                     : '<' . $quad->getObject() . '>';
             } else {
-                $result .= '"' . $quad->getObject()->getValue() . '"';
```
Comment (lanthaler) [inline NQuads.php]: Please change this to

```php
$search = array("\n", '"');
$replace = array('\n', '\"');
$result .= '"' . str_replace($search, $replace, $quad->getObject()->getValue()) . '"';
```

N-Quads supports `\n` for line-feeds, see https://www.w3.org/2013/N-QuadsTests/

File: JsonLD.php (hunk 3 above)
Comment (frmichel) [reply JsonLD.php]: Right, sorry I mixed up several PRs together.

File: FileGetContentsLoader.php (hunk 2 above)
Comment (frmichel) [reply FileGetContentsLoader.php]: Ok. Basically this should not change anything to the default behaviour. But useful to have the line in particular if your want to allow self-signed certificates. 
Anyway I agree this should be in a separate PR.

**Comment by frmichel** [comment]
I'm closing this PR and making 2 separate ones for user agent and escape issue.
